### ISSUE TITLE
Laravel not properly handling sessions if on a subdomain where both are running laravel.

### ISSUE THREAD

**Issue by zocheyado** [laravel/framework#19109]
- Laravel Version: 5.4
- PHP Version: 7.1
- Database Driver & Version: MariaDB 10

So the issue appears to be with the way sessions are handled from the COOKIE_DOMAIN .env variable when you are using DATABASE as the session driver. The issue is that if you have a laravel install on a domain, you have to setup the COOKIE_DOMAIN to match your web domain, this works great, however if you have a subdomain with a totally separate laravel install this causes collisions. Essentially what happens is after you setup your subdomain laravel install you can have the fully qualified domain (subdomain.domain.com) or the higher level domain (domain.com) inside of your .env under COOKIE_DOMAIN and it will work. BUT! if you go to the real domain at any point after that it will destroy your session on the subdomain site, and cause the XSRF and laravel session cookie to NOT save on your browser from that point forward on your subdomain. 

So to recap, 2 laravel installs, 2 databases for sessions and 1 domain name with 1 sub domain will cause collisions between the subdomain and the regular domain based on the COOKIE_DOMAIN variable. No idea if this is chrome issue, or a laravel issue or both but this was the best place I could think to post it.

**Comment by sisve** [comment]
Wouldn't it be enough to rename the default cookie name (in your config/session.php) to something unique per application?

**Comment by zocheyado** [comment]
So making this change DID fix my issue, so thank you for that. However that being said is this a chrome issue or a laravel issue that it cannot differentiate between two cookies with the same name on 2 different sites. It makes sense why it is happening because it is a subdomain. I ended up adding an ENV variable for the cookie so I can switch it on the fly.

**Comment by bbashy** [comment]
@zocheyado It's a configuration issue. If you set your cookie to work on domain + subs, it will break other sites on the subdomains.
